src/Relation/Unary/Closure.agda

```diff
@@ -0,0 +1,72 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Closures of a unary relation with respect to a binary one
+------------------------------------------------------------------------
+
+module Relation.Unary.Closure where
+
+open import Level
+open import Relation.Unary using (Pred)
+open import Relation.Binary as Rel hiding (module Preorder; module StrictPartialOrder)
+
+module Preorder {a r e} (P : Preorder a e r) where
+  open Rel.Preorder P
+
+  -- Depending on whether the order is increasing or decreasing, we get either
+  -- the definition of the upward-closed subsets or that of the downward-closed
+  -- ones.
+  □ : ∀ {t} → Pred Carrier t → Pred Carrier (a ⊔ r ⊔ t)
+  □ T x = ∀ {y} → x ∼ y → T y
```

Comment:
I'm missing the character set for what you've called this type. Any idea what I need to install?

Reply:
It's this character: https://www.fileformat.info/info/unicode/char/25a1/index.htm

We picked it in https://dl.acm.org/citation.cfm?id=3236785 by analogy to modal
logic where box also means "for all successors".

Reply:
It's a box... which might show up looking quite similar [might even be identical] to the box you get if you don't have the font installed. I am not sure how I would debug this!

Reply:
That's exactly the symbol that's used to render characters that aren't installed on my system :laughing: 

Can you think of any other alternatives names? Might be a little confusing to people...

Reply:
The connection to modal logic really calls for box though:

> w ⊨ □ P if and only if for every element u of G, if w R u then u ⊨ P ([source](https://en.wikipedia.org/wiki/Modal_logic))

vs.

> □ T x = ∀ {y} → x ∼ y → T y

Reply:
I think @MatthewDaggitt means can you pick a different box?

Reply:
I guess. Or we need to make it very clear at the top of the file that the user *isn't* missing a load of fonts!

Reply:
Oh, I see. Looking at the input method for the ones available, which one do you prefer?

1. □ -- WHITE SQUARE (the current one)
2. ◻ -- WHITE MEDIUM SQUARE
3. ◽ -- WHITE MEDIUM SMALL SQUARE

Wikipedia seems to advocate either 1. or 2. https://en.wikipedia.org/wiki/List_of_logic_symbols

Reply:
1. and 2. are being rendered identically for me. 3. is noticeably smaller but I guess it's not recommended. 

Hmm I'm not quite sure what to do....

Reply:
> Hmm I'm not quite sure what to do....

I would go with your previous suggestion of adding a preamble at the top
of the file.

It's also a good opportunity to throw in a paragraph or two to give the
reader an intuition of what these mean and clarify the connection with
modal logic, kripke models, etc.